### PR TITLE
Data table documentation

### PR DESCRIPTION
**Pull request by ashjbarnes**
I've added a summary of how to use the `data_table` to the `forcing` page of the docs. I think this might be helpful for new users, as otherwise they need to know to look in the FMS / data_override repository and find the readme there.

### CODE REVIEW

File: docs/forcing.rst [hunk 1]
```diff
@@ -1,5 +1,29 @@
 Forcing
 =======
+Data Override
+-------
+When running MOM6 with the Flexible Modelling System (FMS) coupler, forcing can be specified by a `data_table` file. This is particularly useful when running MOM6 with a data atmosphere, as paths to the relevent atmospheric forcing products (eg. JRA55-do or ERA5) can be provided here. Each item in the data table must be separated by a new line, and contains the following information:
+
+| ``gridname``: The component of the model this data applies to. eg. `atm` `ocn` `lnd` `ice`.
```
Comment (theresa-morrison) [inline docs/forcing.rst:7]: Is this an example of the formatting for a yaml file? I am not too familiar with yaml formatting.

**Review by theresa-morrison — APPROVED** [review]
This is a good addition to our documentation. I had one clarifying comment, but I think this is ready to be added.

File: docs/forcing.rst (hunk 1 above)
Comment (ashjbarnes) [reply docs/forcing.rst:7]: Great thanks Theresa!

> Is this an example of the formatting for a yaml file? I am not too familiar with yaml formatting.

Admittedly I began writing out a yaml example, but I couldn't get the formatting right within the spninx documentation framework. It kept removing the indentations even when I was trying to write plain text blocks. I've also never seen a MOM6 repo that uses the yaml format so perhaps including only the legacy example in the docs is ok? 

Perhaps I could reword to mention that yaml format is also available, check the `data_override` docs for detail

Comment (ashjbarnes) [reply docs/forcing.rst:7]: My last commit hopefully addresses this

Comment (theresa-morrison) [reply docs/forcing.rst:7]: Thanks Ashley this is great!

**Review by theresa-morrison — APPROVED** [review]

**Comment by marshallward** [comment]
Thank you @ashjbarnes, someone within GFDL just asked me for something like this a few weeks ago.  I think this will certainly be appreciated by others.

**Review by marshallward — APPROVED** [review]

**Pull request merged**